src/blocking/spi.rs

```diff
@@ -104,3 +104,31 @@ pub mod write_iter {
         }
     }
 }
+
+
+/// Actions for transactional SPI trait
+/// 
+/// This allows composition of SPI operations into a single bus transaction
+#[cfg(feature = "unproven")]
+#[derive(Debug, PartialEq)]
+pub enum Actions<'a> {
+    /// Read data into the provided buffer, write data undefined
+    Read(&'a mut [u8]),
```

Comment:
Is there a concrete advantage to having `Read` in addition to `Transfer`? In both cases, you have to create and initialize a mutable buffer -- at least if you're calling from safe code. But I could see "write data undefined" being a low-caliber footgun. Maybe you _do_ care about the outgoing bytes, but the HAL implementation you're initially testing with happens to do the right thing.

What about `Read(u8, &'a mut [u8])`, where the first byte is clocked out repeatedly for the duration of the transaction?

Reply:
> Is there a concrete advantage to having Read in addition to Transfer? 

Some devices support read operations without transferring data out to the peripheral.

> I could see "write data undefined" being a low-caliber footgun

Yep, unfortunately when _only_ reading neither [spidev](https://github.com/spotify/linux/blob/master/include/linux/spi/spidev.h) or the [CP2130](https://www.silabs.com/documents/public/application-notes/AN792.pdf) device I've been playing with support configuring what data is shifted out.

As it's hardware dependent, I don't think this is in scope for the hal. Would a more explicit warning on the `Read` operation mitigate this concern?

Reply:
> Alternately Write and Transfer would cover most use cases, so, Read could just be dropped.

I would be in favor of that